modules/apps/data-engine/data-engine-js-components-web/src/main/resources/META-INF/resources/js/utils/rulesSupport.js

```diff
@@ -373,11 +373,22 @@ export const isActionsValid = (actions) =>
 				case 'auto-fill': {
 					const {inputs, outputs} = payload;
 
-					return (
-						Boolean(target) &&
-						(Boolean(Object.values(inputs).length) ||
-							Boolean(Object.values(outputs).length))
+					let valid =
+						Boolean(Object.values(inputs).length) ||
+						Boolean(Object.values(outputs).length);
+
+					const currentDataProvider = dataProviders.filter(
```

Comment:
Since you are not expecting multiples `dataProviders` in response to that operation, use [find](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) instead (it will stop the iteration at first result found returning just the item you need):
```suggestion
					const currentDataProvider = dataProviders.find(
```